test/unit/lexer/location_offset.d

```diff
@@ -0,0 +1,489 @@
+    "class_" : Test("class"),
+    "interface_" : Test("interface"),
+    "union_" : Test("union"),
+    "enum_" : Test("enum"),
+    "import_" : Test("import"),
+    "alias_" : Test("alias"),
+    "override_" : Test("override"),
+    "delegate_" : Test("delegate"),
+    "function_" : Test("function"),
+    "mixin_" : Test("mixin"),
+    "align_" : Test("align"),
+    "extern_" : Test("extern"),
+    "private_" : Test("private"),
+    "protected_" : Test("protected"),
+    "public_" : Test("public"),
+    "export_" : Test("export"),
+    "static_" : Test("static"),
+    "final_" : Test("final"),
+    "const_" : Test("const"),
+    "abstract_" : Test("abstract"),
+    "debug_" : Test("debug"),
+    "deprecated_" : Test("deprecated"),
+    "in_" : Test("in"),
+    "out_" : Test("out"),
+    "inout_" : Test("inout"),
+    "lazy_" : Test("lazy"),
+    "auto_" : Test("auto"),
+    "package_" : Test("package"),
+    "immutable_" : Test("immutable"),
+
+    "if_" : Test("if"),
+    "else_" : Test("else"),
+    "while_" : Test("while"),
+    "for_" : Test("for"),
+    "do_" : Test("do"),
+    "switch_" : Test("switch"),
+    "case_" : Test("case"),
+    "default_" : Test("default"),
+    "break_" : Test("break"),
+    "continue_" : Test("continue"),
+    "with_" : Test("with"),
+    "synchronized_" : Test("synchronized"),
+    "return_" : Test("return"),
+    "goto_" : Test("goto"),
+    "try_" : Test("try"),
+    "catch_" : Test("catch"),
+    "finally_" : Test("finally"),
+    "asm_" : Test("asm"),
+    "foreach_" : Test("foreach"),
+    "foreach_reverse_" : Test("foreach_reverse"),
+    "scope_" : Test("scope"),
+
+    "invariant_" : Test("invariant"),
+
+    "unittest_" : Test("unittest"),
+
+    "argumentTypes" : Test("__argTypes"),
+    "ref_" : Test("ref"),
+    "macro_" : Test("macro"),
+
+    "parameters" : Test("__parameters"),
+    "traits" : Test("__traits"),
+    "overloadSet" : Test("__overloadset"),
+    "pure_" : Test("pure"),
+    "nothrow_" : Test("nothrow"),
+    "gshared" : Test("__gshared"),
+
+    "line" : Test("__LINE__"),
+    "file" : Test("__FILE__"),
+    "fileFullPath" : Test("__FILE_FULL_PATH__"),
+    "moduleString" : Test("__MODULE__"),
+    "functionString" : Test("__FUNCTION__"),
+    "prettyFunction" : Test("__PRETTY_FUNCTION__"),
+
+    "shared_" : Test("shared"),
+    "at" : Test("at sign", "@"),
+    "pow" : Test("power", "^^"),
+    "powAssign" : Test("power assign", "^^="),
+    "goesTo" : Test("fat arrow", "=>"),
+    "vector" : Test("__vector"),
+    "pound" : Test("pound", "#"),
+];
+
+// Ignore tokens not produced by the lexer or tested above
+enum ignoreTokens
+{
+    reserved,
+    negate,
+    array,
+    call,
+    address,
+    star,
+    type,
+    dotVariable,
+    dotIdentifier,
+    dotTemplateInstance,
+    dotType,
+    symbolOffset,
+    variable,
+    arrayLength,
+    dotTemplateDeclaration,
+    declaration,
+    dSymbol,
+    uadd,
+    remove,
+    newAnonymousClass,
+    comment,
```

Comment:
This is produced by the lexer if `commentToken` in the constructor is set to `true`.